### ISSUE TITLE
Keys in configuration file not respected

### ISSUE THREAD

**Issue by Tastaturtaste** [EmbarkStudios/cargo-deny-action#81]
**Describe the bug**
cargo-deny-action does not respect keys `graph.exclude-unpublished` and `graph.exclude-dev` in configuration file

**To Reproduce**
1. Fork https://github.com/Tastaturtaste/argmin
2. Run workflow `argmin CI` on branch `remove_cargo_deny_exception`

**Expected behavior**
The same output as I get when running `cargo deny --all-features --log-level warn check licenses bans sources`

**Screenshots**
See [this log](https://github.com/Tastaturtaste/argmin/actions/runs/11955547809/job/33328172996#step:4:11) and compare with a run of `cargo deny check licenses bans sources` in a local checkout of branch `remove_cargo_deny_exception`

**Additional context**
This bug was detected while working on https://github.com/argmin-rs/argmin/pull/386.

**Comment by Jake-Shadle** [comment]
They are respected. You're not seeing errors because you presumably have a Cargo.lock with older versions of various crates, if you `cargo update` to get the lockfile that is generated in that project's CI, you'll see the same errors locally. I cloned and checked out your PR and get the same errors seen in your CI run.

```
cargo-deny [main●] % cargo run -- --manifest-path ../temp-clones/argmin/Cargo.toml -L off check -s                                                                                                                                       130 ↵
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.17s
     Running `target/debug/cargo-deny --manifest-path ../temp-clones/argmin/Cargo.toml -L off check -s`
[src/cargo-deny/common.rs:151:16] self.exclude_dev = true
[src/cargo-deny/common.rs:173:12] self.exclude_unpublished = true
 advisories FAILED: 3 errors, 0 warnings, 0 notes
           bans ok: 0 errors, 42 warnings, 0 notes
   licenses FAILED: 20 errors, 1 warnings, 448 notes
        sources ok: 0 errors, 0 warnings, 0 notes
```

**Comment by Tastaturtaste** [comment]
You are right. I am sorry to have caused confusion and forgot to check this possibility. Thank you for the fast response!
